src/types/stardust/block/payload/mod.rs

```diff
@@ -20,16 +21,22 @@ pub use self::{
 };
 use crate::types::context::{TryFromWithContext, TryIntoWithContext};
 
+/// The different payloads of a [`Block`](crate::types::stardust::block::Block).
 #[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
 #[serde(rename_all = "snake_case", tag = "kind")]
 pub enum Payload {
+    /// Signals a [`TransactionPayload`].
     Transaction(Box<TransactionPayload>),
+    /// Signals a [`MilestonePayload`].
     Milestone(Box<MilestonePayload>),
+    /// Signals a [`TreasuryTransactionPayload`].
     TreasuryTransaction(Box<TreasuryTransactionPayload>),
+    /// Signals a [`TaggedDataPayload`].
     TaggedData(Box<TaggedDataPayload>),
 }
 
 impl Payload {
+    /// Returns the kind of a payload as a [`&str`](str).
```

Comment:
Seems a little weird. This is pretty "internal" so I don't know why we should bother documenting it. So maybe it should be pub(crate).